### PR TITLE
introduce @effect/schema

### PR DESCRIPTION
**Pull request by KhraksMamtsov**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,7 +30,9 @@
     "@deepkit/core": "1.0.1-alpha.147",
     "@deepkit/type": "1.0.1-alpha.151",
     "@deepkit/type-compiler": "1.0.1-alpha.150",
+    "@effect/schema": "^0.67.12",
```
Comment (moltar) [inline package.json:33]: Please use a pinned version. Renovate will keep it up to date.

**Review by moltar — CHANGES_REQUESTED** [review]
Also, please add to the README (a-z).

File: package.json (hunk 1 above)
Comment (KhraksMamtsov) [reply package.json]: done

**Review by moltar — APPROVED** [review]

**Comment by moltar** [comment]
screenshot-20240527T145638-TMrKagW5@2x


Do we need the result const?

**Comment by KhraksMamtsov** [comment]
Nope

**Comment by KhraksMamtsov** [comment]
But otherwise it shows an error
```
Assertions require every name in the call target to be declared with an explicit type annotation.ts(2775)

effect-schema.ts(67, 9): 'asserts' needs an explicit type annotation.
```

**Comment by moltar** [comment]
> But otherwise it shows an error
> 
> ```
> Assertions require every name in the call target to be declared with an explicit type annotation.ts(2775)
> 
> effect-schema.ts(67, 9): 'asserts' needs an explicit type annotation.
> ```

Where do you seed that? 

PR build seems to pass.

**Comment by KhraksMamtsov** [comment]
I fixed this warning)
when can I expect a merge?

**Comment by moltar** [comment]
I was waiting for @hoeck to review. But maybe he's busy. I'll merge for now.

@hoeck Please do a reverse review when/if you have time.

**Comment by hoeck** [comment]
@moltar had already a look at it and was waiting for the explanation for the warning that got fixed.

Looks all good to me :+1: 

Thanks @KhraksMamtsov

**Pull request merged**